resources/js/Components/Form/CampaignRegistrationForm/CampaignRegistrationForm.tsx

```diff
@@ -0,0 +1,64 @@
+import {useForm} from '@inertiajs/inertia-react'
+import {Checkbox, FormControlLabel, Grid, Typography} from '@mui/material'
+import {ErrorText, Select} from 'Components'
+import React from 'react'
+import {CampaignData} from 'Types/campaign-data'
+import {CharacterData} from 'Types/character-data'
+
+type CampaignRegistrationFormPropType = {
+    campaign: CampaignData
+    characters: CharacterData[]
+}
+
+type CampaignRegistrationFormDataType = {
+    character_id: number | null
+}
+
+const CampaignRegistrationForm = ({
+    campaign,
+    characters,
+}: CampaignRegistrationFormPropType) => {
+    const CAMPAIGN_REGISTRATION_FORM_INITIAL_VALUE: CampaignRegistrationFormDataType = {
```

Comment:
no point of declaring a constant if it's only going to be used once or no conditional data